site-src/geps/gep-1364.md

```diff
@@ -0,0 +1,409 @@
+for bounded exection.
+
+From these guidelines, we can see that Conditions can be either _positive polarity_
+(healthy resources have them as `status: true`) or _negative polarity_ (healthy
+resources have them as `status: false`). `Ready` is an example of a positive polarity
+condition, and conditions like `Conflicted` from Listener or `NetworkUnavailable`,
+`MemoryPressure`, or `DiskPressure` from the Node resource are examples of
+negative-polarity conditions.
+
+There is also some extra context that's not in the API conventions doc:
+
+SIG-API Machinery has been reluctant to add fields that would aid in machine-parsing
+of Conditions, especially fields that would indicate the polarity, because they
+are intended more for human consumption than machine consumption.
+
+This means that there's no guidance from upstream about condition polarity. We'll
+discuss this more when we talk about new conditions.
+
+The guidance about Conditions being added as soon as a controller sees a resource
+is a bit unclear - as written in the conventions, it seems to imply that _all_ 
+relevant conditions should always be added, even if their status has to be set to
+`unknown`.
+Gateway API resources do not currently require this, and the practice seems to be
+uncommon.
+
+## Proposed changes
+
+### Proposed changes summary
+
+* All the current Conditions that indicate that the resource is okay and ready
+for processing witll be replaced with `Attached`, except for GatewayClass (since
+it is the root of the resource tree, it will stay with `Accepted`).
+* In general, resources should be considered `Attached` if their config is valid
+enough to generate some config in the underlying data plane.
+* There will be a limited set of positive polarity summary conditions, and a number
+of other specific negative-polarity error conditions.
+* All relevant positive-polarity summary Conditions for a resource must be added
+when it's observed.
+For example, HTTPRoutes must always have `Accepted` and `ResolvedRefs`, regardless
+of their state.
+* Negative polarity error conditions must only be added when the error is True.
+* The `Ready` condition will be moved to Extended conformance, and we'll reevaluate
+if it's used by any implementations after some time has passed. If not, it may be
+removed.
+* To capture the behavior that `Ready` currently captures, `Programmed` will be
+introduced. This means that the implementation has seen the config, has everything
+it needs, parsed it, and sent configuration off to the data plane. The configuration
+should be available "soon". We'll leave "soon" undefined for now.
+* The summary conditions will be added into the CRD definitions
+as default values for `status.Conditions`, with a status of `Unknown`.
+* Resolving a comment that came up, documentation will be added to clarify that
+it's okay to add your own Conditions, and that implementations should namespace
```

Comment:
We should probably also recommend that implementations publish both new and old conditions for several months to provide a smoother transition.